pkg/multiterm/virtualterm.go

```diff
@@ -0,0 +1,44 @@
+package multiterm
+
+import (
+	"fmt"
+	"io"
+	"sort"
+)
+
+type VirtualTerm struct {
+	lines map[int]string
+}
+
+func NewVirtualTerm() *VirtualTerm {
```

Comment:
Needs interface assertion, as well as some better naming below. eg `LineCount` is not technically correct.